Sources/SwiftProtobuf/TextScanner.swift

```diff
@@ -75,206 +247,525 @@ public class TextScanner {
-            case "a"..."z", "A"..."Z", "0"..."9", "_":
-                index = scalars.index(after: index)
+            case 97...122, 65...90, 48...57, 95:
+                index = utf8.index(after: index)
             default:
-                return String(scalars[tokenStart..<index])
+                return String(utf8[start..<index])
             }
         }
-        return String(scalars[tokenStart..<index])
+        return String(utf8[start..<index])
     }
-    
+
     /// Parse the rest of an [extension_field_name] in the input, assuming the
     /// initial "[" character has already been read (and is in the prefix)
-    /// This is also used in Any for the typeURL, so we include "/", "."
-    private func parseExtensionIdentifier() -> String? {
-        if index == scalars.endIndex {
+    /// This is also used for AnyURL, so we include "/", "."
+    private func parseExtensionKey() -> String? {
+        let start = index
+        if index == utf8.endIndex {
             return nil
         }
-        let c = scalars[index]
+        let c = utf8[index]
         switch c {
-        case "a"..."z", "A"..."Z":
-            index = scalars.index(after: index)
+        case 97...122, 65...90:
+            index = utf8.index(after: index)
         default:
             return nil
         }
-        while index != scalars.endIndex {
-            let c = scalars[index]
+        while index != utf8.endIndex {
+            let c = utf8[index]
             switch c {
-            case "a"..."z", "A"..."Z", "0"..."9", "_", ".", "/":
-                index = scalars.index(after: index)
-            case "]":
-                index = scalars.index(after: index)
-                return String(scalars[tokenStart..<index])
+            case 97...122, 65...90, 48...57, 95, 46, 47:
+                index = utf8.index(after: index)
+            case 93: // ]
+                let s = String(utf8[start..<index])
+                return s
             default:
                 return nil
             }
         }
         return nil
     }
-    
+
     /// Assumes the leading quote has already been consumed
-    private func parseQuotedString(terminator: UnicodeScalar) -> String? {
-        tokenStart = index
-        while index != scalars.endIndex {
-            let c = scalars[index]
+    private func parseQuotedString(terminator: UInt8) -> String? {
+        let start = index
+        while index != utf8.endIndex {
+            let c = utf8[index]
             if c == terminator {
-                let s = String(scalars[tokenStart..<index])
-                index = scalars.index(after: index)
+                let s = String(utf8[start..<index])
+                index = utf8.index(after: index)
                 return s
             }
-            index = scalars.index(after: index)
-            if c == "\\" {
-                if index == scalars.endIndex {
+            index = utf8.index(after: index)
+            if c == 92 { //  \
```

Comment:
nit: maybe // '\' for these sorta comments, to make it clear the character you are talking about?